Spec.md

```diff
@@ -0,0 +1,223 @@
+# OPC UA Pub/Sub Interface Description
+
+This document defines the Interface of the umati Dashboard based on OPC UA PubSub.
+It describes how a OPC UA server address space must be mapped to OPC UA PubSub, so that the infromation can map to the templates and displayed.
+
+## Requirements for the Mapping of the address space to PubSub
+
+(Reference: [umati/UA-CloudPublisher#5](https://github.com/umati/UA-CloudPublisher/issues/5))
+
+- Ability to react to dynamic changes in the address space (nodes added/deleted)
+  - Changes should be detected by monitoring NodeManagement events.
+- Resolve references to nodes (NodeId as values)
+- Hierarchy references must be mapped
+- Non-hierarchy references should be mapped if necessary
+  - Non-hierarchy references are mapped if relevant to the data semantics.
+- Provide type information
+- Methods should be considered
+- Online detection (possibly as separate topics)
+- Allow empty objects
+- Events/alarms must be mappable
+- Coordinate the completeness of the run
+- Encoding should be UA JSON
+- Dimensioning for approximately 350 subscribers (machines + dashboard users)
+- Custom DataTypes must be included
+- The address space includes at least OPC UA Machinery and possibly other Companion Specifications based on it.
+- The entry point of the mapping is the machine object in the Machines folder of Machinery.
+
+## Design Decisions
+
+The concept of OPC UA Part 14 should be followed as closely as possible.
+
+## Mapping
+
+### Mapping of Objects and Variables
+
+- An object is mapped to a DataSet.
+- A variable is mapped as a field to the DataSet. If the variable contains children (other variables), a DataSet with the children is created.
+
+#### Example
+
+![image](https://github.com/umati/UA-CloudPublisher/assets/12342602/91eae1d7-9d6f-4246-abb2-0ec5ea9b59bc)
+
+The image shows a part of an address space. Each object is mapped to a DataSet:
+
+- Production
+- ActiveProgram
+- State
+
+The properties Name, NumberInList, Id, and Number are fields of the DataSets. The value of CurrentState is mapped as a field of ActiveProgram and has a DataSet with its properties:

```

Comment:
I think it would be more generic, if CurrentState would be mapped to the  State and not to to the second-level node (ActiveProgramm) in the hierarchy.

Reply:
Yes that is correct. I have don it so at the example.